modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1495,3 +1495,68 @@ def rebalance_partitions(cls, partitions):
             lengths = [part.length() for part in new_partitions[:, 0]]
             return new_partitions, lengths
         return partitions, None
+
+    @classmethod
+    def shuffle_partitions(
+        cls, partitions, index, sample_func, pivot_func, split_func, final_shuffle_func
+    ):
+        """
+        Return shuffled partitions.
+
+        Parameters
+        ----------
+        partitions : np.ndarray
+            The 2-d array of partitions to shuffle.
+        index : int
+            The index of partitions corresponding to the partitions that contain the column to sample.
+        sample_func : Callable(pandas.DataFrame) -> np.ndarray
+            Function to sample partitions. Output of this function will be passed to ``pivot_func``
+            to determine pivots.
+        pivot_func : Callable(List[Any]) -> np.ndarray
+            Function to determine pivots from partitions. The pivots determined by this function
+            will be passed to ``split_func`` to split each partition.
+        split_func : Callable(pandas.DataFrame, Any) -> List[pandas.Dataframe]
+            Function that splits a partition based off of the pivots provided. Should return an
+            unpacked list of pandas Dataframes.
+        final_shuffle_func : Callable(pandas.DataFrame) -> pandas.DataFrame
+            Function that shuffles the data within each new partition.
+
+        Returns
+        -------
+        np.ndarray
+            A list of row-partitions that have been shuffled.
+        """
+        # Mask the partition that contains the column that will be sampled.
+        masked_partitions = partitions[:, index]
+        # Sample each partition
+        sample_func = cls.preprocess_func(sample_func)
+        samples = [partition.apply(sample_func) for partition in masked_partitions]
+        # Get each sample to pass in to the pivot function
+        samples = [sample.get() for sample in samples]
```

Comment:
I think we have a "parallel get" method in our partition manager for that? cc @anmyachev